pilot/pkg/serviceregistry/external/servicediscovery.go

```diff
@@ -534,11 +641,43 @@ func (d *ServiceEntryStore) GetProxyWorkloadLabels(proxy *model.Proxy) (labels.C
 	return out, nil
 }
 
-// GetIstioServiceAccounts implements model.ServiceAccounts operation TODOg
-func (d *ServiceEntryStore) GetIstioServiceAccounts(*model.Service, []int) []string {
-	//for service entries, there is no istio auth, no service accounts, etc. It is just a
-	// service, with service instances, and dns.
-	return nil
+// GetIstioServiceAccounts implements model.ServiceAccounts operation
+// For service entries using workload entries or mix of workload entries and pods,
+// this function returns the appropriate service accounts used by these.
+func (d *ServiceEntryStore) GetIstioServiceAccounts(svc *model.Service, ports []int) []string {
+	// service entries with built in endpoints have SANs as a dedicated field.
```

Comment:
Please note: this is a bug fix. This function should have been implemented (actually it was implemented IIRC) but somehow it got nuked. ServiceEntries have SAN fields. And workloads have service account fields. These should be processed. Otherwise, istio mTLS will break when talking to service entries/workload entries.

Reply:
Good catch !